main.py

```diff
@@ -0,0 +1,322 @@
+                             and filter_gc(record, gc_bounds)
+                             and filter_quality(record, quality_threshold))
+    SeqIO.write(filtered_seq_iterator, output_path, "fastq")
+
+
+def filter_length(record: SeqRecord, length_bounds: tuple) -> bool:
+    return length_bounds[0] <= len(record.seq) <= length_bounds[1]
+
+
+def filter_gc(record: SeqRecord, gc_bounds: tuple) -> bool:
+    return gc_bounds[0] <= GC123(record.seq)[0] <= gc_bounds[1]
+
+
+def filter_quality(record: SeqRecord, quality_threshold: int) -> bool:
+    avg_quality = np.mean(record.letter_annotations["phred_quality"])
+    return avg_quality >= quality_threshold
+
+
+# -----------------------Biological sequence classes----------------------------
+class BiologicalSequence(ABC, str):
+    @abstractmethod
+    def check_alphabet(self) -> bool:
+        pass
+
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Class representing nucleic acid sequences.
+
+    Attributes:
+        ALPHABET (Tuple[str, ...])
+        COMPLEMENT_DICT (Dict[str, str])
+
+    Methods:
+        - check_alphabet: Checks if the provided nucleic acid sequence contains
+        only acceptable letters
+        - complement: Complements DNA or RNA sequence
+        - get_GC: Calculates GC content for nucleic acid sequence
+
+    """
+    ALPHABET: Tuple[str, ...]
+    COMPLEMENT_DICT: Dict[str, str]
+
+    def __init__(self, sequence):
+        raise NotImplementedError('An instance of this class cannot be created')
+
+    def check_alphabet(self) -> bool:
+        """Checks if the provided nucleic acid sequence contains
+        only acceptable letters"""
+        sequence_chars = set(self.sequence)
+        return sequence_chars.issubset(type(self).ALPHABET)
+
+    def complement(self) -> 'NucleicAcidSequence':
+        """Complements DNA or RNA sequence."""
+        complement_seq = ''.join(type(self).COMPLEMENT_DICT.get(base)
+                                 for base in self.sequence)
+        return type(self)(complement_seq)
+
+    def get_gc(self) -> float:
+        """Calculates GC content for nucleic acid sequence"""
+        return GC123(self.sequence)[0]
+
+    def __repr__(self):
+        """Returns representation of the object"""
+        return f"{self.__class__.__name__}: {self.sequence}"
+
+    def __str__(self):
+        return self.sequence
+
+
+class RNASequence(NucleicAcidSequence):
+    """
+    Class representing RNA sequences
+
+    Attributes:
+        - ALPHABET: contains valid for RNA characters
+        - COMPLEMENT_DICT: contains rules for complement
+
+    Methods: refer to parent class 'NucleicAcidSequence'
+        """
+    ALPHABET = ('A', 'U', 'G', 'C', 'N')
```

Comment:
```suggestion
    ALPHABET = {'A', 'U', 'G', 'C', 'N'}
```

Разница не большая, но зато по сути так прям то что надо